src/chainstate/burn/sortition.rs

```diff
@@ -89,15 +90,22 @@ impl BlockSnapshot {
             canonical_stacks_tip_height: 0,
             canonical_stacks_tip_hash: FIRST_STACKS_BLOCK_HASH.clone(),
             canonical_stacks_tip_consensus_hash: FIRST_BURNCHAIN_CONSENSUS_HASH.clone(),
+            // Initial snapshot sets sortition_id = burn_header_hash,
+            //  we shouldn't need to update this to use PoxId::initial(),
+            //  but if we do, we need to update a lot of test cases.
```

Comment:
I guess I'm having a hard time seeing this right now.  How is `SortitionId` supposed to be calculated, if it's not the burn header hash?

Reply:
In `chainstate::burn::db::processing`, before creating the BlockSnapshot, the `next_sortition_id` is calculated, using the parent's PoX id -- that parent PoxId is fetched during `evaluate_sortition` (from the MARF entry for the "current" PoxId).

Looking at that code, there may be a off-by-one error there. I think that in db::processing, it may need to compute the sortition_id from the parent pox + extend it with the current reward cycle info if set, but I would want to create a test case to confirm that once we have the ability to create a PoX fork before changing anything there.

Reply:
Did you have a chance to look more onto this possible off-by-one error?

Reply:
Yep— there’s indeed an off-by-one in the way it computes the next PoX ID. It won’t create any observable bugs in this PR, since there are no possible PoX forks in this PR. The anchor block detection PR will fix that bug (the test for PoX fork handling revealed it)

Reply:
Backported the fix from `feat/anchor-block-detect`